libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloClient.kt

```diff
@@ -150,19 +152,41 @@ private constructor(
 
     return DefaultInterceptorChain(interceptors + networkInterceptor, 0)
         .proceed(request)
-        .let {
+        .let { flow ->
           if (request.useV3ExceptionHandling == true) {
-            it.onEach { response ->
+            flow.onEach { response ->
               if (response.exception != null) {
                 throw response.exception!!
               }
             }
           } else {
-            it
+            flow
+          }
+        }
+        .let { flow ->
+          if (ignorePartialData == true) {
+            flow.map { response ->
+              if (response.data != null && response.hasErrors()) {
+                response.withNullData()
+              } else {
+                response
+              }
+            }
+          } else {
+            flow
           }
         }
   }
 
+  private fun <D : Operation.Data> ApolloResponse<D>.withNullData() =
+      ApolloResponse.Builder(operation = operation, requestUuid = requestUuid, data = null)
+          .errors(errors)
+          .exception(exception)
+          .extensions(extensions)
+          .addExecutionContext(executionContext)
+          .isLast(isLast)
+          .build()
```

Comment:
Don't we have `ApolloResponse.newBuilder()`? If we don't we should add. it

Reply:
There's one but you can't change `data` with it 😅 But I can add it actually.